pkg/oc/admin/diagnostics/diagnostics/systemd/analyze_logs.go

```diff
@@ -99,8 +99,13 @@ func (d AnalyzeLogs) Check() types.DiagnosticResult {
 					if err := json.Unmarshal(bytes, &entry); err != nil {
 						r.Debug("DS0003", fmt.Sprintf("Couldn't read the JSON for this log message:\n%s\nGot error %s", string(bytes), errStr(err)))
 					} else {
-						if lineCount > 500 && stampTooOld(entry.TimeStamp, timeLimit) {
-							r.Debug("DS0004", fmt.Sprintf("Stopped reading %s log: timestamp %s too old", unitName, entry.TimeStamp))
+						old, unixTimestamp, err := stampTooOld(entry.TimeStamp, timeLimit)
```

Comment:
I have a few suggestions to make here.

`stampTooOld` is only ever used here. All it really accomplishes is making this logic easy to understand. Here you've turned the method into trinary logic (true/false/error) plus return value and then you have to process that trinary again upon receiving it. I think it would be simpler just to bring the method inline and use loop controls rather than `return`.